lib/place_v7.ml

```diff
@@ -0,0 +1,659 @@
+  if len = 0 then nosub ""
+  else if String.unsafe_get s 0 = '[' then
+    match String.index_opt s ']' with
+    | None -> nosub s
+    | Some i -> (
+        match
+          let rec loop b i =
+            if i = len then None
+            else
+              match Char.code s.[i] with
+              | 0x20 -> loop b (i + 1)
+              | 0x2D when not b -> loop true (i + 1) (* hyphen *)
+              (* handle en and em dash as well *)
+              | 0xE2
+                when Char.code s.[i + 1] = 0x80
+                     && (Char.code s.[i + 2] = 0x93
+                        || Char.code s.[i + 2] = 0x94)
+                     && not b ->
+                  loop true (i + 3)
+              | _ -> if b then Some i else None
+          in
+          loop false (i + 1)
+        with
+        | None -> nosub s
+        | Some j -> sub s len i j)
+  else nosub s
+
+(** [split_suburb "[foo-bar] - boobar (baz)"] is ["foo-bar", "boobar (baz)")] *)
+let split_suburb =
+  suburb_aux
+    (fun s len i j -> (String.sub s 1 (i - 1), String.sub s j (len - j)))
+    (fun s -> ("", s))
+
+(** [only_suburb "[foo-bar] - boobar (baz)"] is ["foo-bar"]
+    [only_suburb "boobar (baz)"] is [""] *)
+let only_suburb =
+  suburb_aux (fun s _len i _j -> String.sub s 1 (i - 1)) (fun _ -> "")
+
+(** [without_suburb "[foo-bar] - boobar (baz)"] is ["boobar (baz)"]
+    [without_suburb "boobar (baz)"] is ["boobar (baz)"] *)
+let without_suburb =
+  suburb_aux (fun s len _i j -> String.sub s j (len - j)) (fun s -> s)
+
+let has_suburb s = String.unsafe_get s 0 = '['
+
+type 'a env =
+  | Vlist_data of (string * (string * int) list) list
+  | Vlist_ini of string list
+  | Vlist_value of (string * (string * int) list) list
+  | Venv_keys of (string * int) list
+  | Vint of int
+  | Vstring of string
+  | Vbool of bool
+  | Vother of 'a
+  | Vnone
+
+let get_vother = function Vother x -> Some x | _ -> None
+let set_vother x = Vother x
+
+let normalize =
+  suburb_aux
+    (fun s len i j ->
+      let b = Bytes.create (len - j + i + 1) in
+      Bytes.blit_string s 1 b 0 (i - 1);
+      Bytes.unsafe_set b (i - 1) ',';
+      Bytes.unsafe_set b i ' ';
+      Bytes.blit_string s j b (i + 1) (len - j);
+      Bytes.unsafe_to_string b)
+    (fun s -> s)
+
+let compare_places s1 s2 =
+  let ss1, s1 = split_suburb s1 in
+  let ss2, s2 = split_suburb s2 in
+  match
+    Mutil.list_compare Gutil.alphabetic_order
+      (String.split_on_char ',' s1)
+      (String.split_on_char ',' s2)
+  with
+  | 0 -> Gutil.alphabetic_order ss1 ss2
+  | x -> x
+
+let max_rlm_nbr conf =
+  match p_getenv conf.env "max_rlm_nbr" with
+  | Some n ->
+      if n = "" then
+        match List.assoc_opt "max_rlm_nbr" conf.base_env with
+        | Some n -> if n = "" then 80 else int_of_string n
+        | None -> 80
+      else int_of_string n
+  | None -> (
+      match List.assoc_opt "max_rlm_nbr" conf.base_env with
+      | Some n -> if n = "" then 80 else int_of_string n
+      | None -> 80)
```

Comment:
can you define a local variable `let default_line_length = 80`?
what does rlm stand for?